### PR TITLE
Indigo dev

### PR DESCRIPTION
**Pull request by ipa-jsf**
indigo_dev build.
Merged changes from hydro_dev.
Successfully build and catkin_lint it
@ipa-fxm Please Review

### CODE REVIEW

**Comment by fmessmer** [comment]
see my comments in [cob_perception_common#41](https://github.com/ipa320/cob_perception_common/pull/41)

File: cob_fiducials/common/src/AbstractFiducialModel.cpp [hunk 1]
```diff
@@ -0,0 +1,254 @@
+#include "../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/AbstractFiducialModel.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/FiducialTestingEnvironment.cpp [hunk 2]
```diff
@@ -1,16 +1,18 @@
-//#include "../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/FiducialTestingEnvironment.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/FiducialModelAruco.cpp [hunk 3]
```diff
@@ -1,11 +1,14 @@
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/FiducialModelAruco.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/arucofidmarkers.cpp [hunk 4]
```diff
@@ -28,14 +28,14 @@ or implied, of Rafael Muñoz Salinas.
 
 */
 
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/arucofidmarkers.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/board.cpp [hunk 5]
```diff
@@ -25,14 +25,17 @@ The views and conclusions contained in the software and documentation are those
 authors and should not be interpreted as representing official policies, either expressed
 or implied, of Rafael Muñoz Salinas.
 ********************************/
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/board.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/boarddetector.cpp [hunk 6]
```diff
@@ -25,19 +25,24 @@ The views and conclusions contained in the software and documentation are those
 authors and should not be interpreted as representing official policies, either expressed
 or implied, of Rafael Muñoz Salinas.
 ********************************/
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/boarddetector.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/cameraparameters.cpp [hunk 7]
```diff
@@ -25,16 +25,19 @@ The views and conclusions contained in the software and documentation are those
 authors and should not be interpreted as representing official policies, either expressed
 or implied, of Rafael Muñoz Salinas.
 ********************************/
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/cameraparameters.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/cvdrawingutils.cpp [hunk 8]
```diff
@@ -25,9 +25,11 @@ The views and conclusions contained in the software and documentation are those
 authors and should not be interpreted as representing official policies, either expressed
 or implied, of Rafael Muñoz Salinas.
 ********************************/
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/cvdrawingutils.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/marker.cpp [hunk 9]
```diff
@@ -26,15 +26,19 @@ authors and should not be interpreted as representing official policies, either
 or implied, of Rafael Muñoz Salinas.
 ********************************/
 
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/marker.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

File: cob_fiducials/common/src/aruco/markerdetector.cpp [hunk 10]
```diff
@@ -25,18 +25,20 @@ The views and conclusions contained in the software and documentation are those
 authors and should not be interpreted as representing official policies, either expressed
 or implied, of Rafael Muñoz Salinas.
 ********************************/
-//#include "../../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../../cob_object_perception/common/include/cob_object_perception/StdAfx.h"
```
Comment (fmessmer) [inline cob_fiducials/common/src/aruco/markerdetector.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

**Comment by fmessmer** [comment]
More review to come after the first issues have been adressed...
Should not be merged like this!

**Comment by ipa-jsf** [comment]
@ipa-fxm Fixed issue with precompiled header files

File: libzxing/Makefile.tarball [hunk 11]
```diff
@@ -1,8 +1,11 @@
 all: installed
 
 TARBALL = build/zxing-2.1.tar.gz
-TARBALL_URL = https://code.ros.org/svn/release/download/thirdparty/zxing-2.1.tar.gz # originally from http://zxing.googlecode.com/files/ZXing-2.1.zip
+TARBALL_URL = zxing-2.1.tar.gz # Tarball from code.ros.org no longer compiling on ubuntu 14.04
+#https://code.ros.org/svn/release/download/thirdparty/zxing-2.1.tar.gz 
+# originally from http://zxing.googlecode.com/files/ZXing-2.1.zip
```
Comment (fmessmer) [inline libzxing/Makefile.tarball:5]: Well, is there a new version for the zxing lib for 14.04?

I think we should not add and commit the actual tarball into the code repo?
Maybe we need to host the tarball on our own repo somewhere? 
@ipa-fmw ?

**Comment by fmessmer** [comment]
I also get the following Warning for libzxing:

```
-- +++ processing catkin package: 'libzxing'
-- ==> add_subdirectory(cob_object_perception/libzxing)
-- Copying /home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/libzxing/common/lib/libzxing.a to /home/fxm/projects/indigo/care-o-bot/devel/lib
You have called ADD_LIBRARY for library zxing without any source files. This typically indicates a problem with your CMakeLists.txt file
```

But I think that's the same Warning that we get for the packages in `cob_extern`
Any solution for this? @ipa-fmw?

**Comment by fmessmer** [comment]
Please fix the deprecation warnings....No mercy this time! :smile: 

```
Scanning dependencies of target cob_read_text
[ 84%] Building CXX object cob_object_perception/cob_read_text/CMakeFiles/cob_read_text.dir/ros/src/cob_read_text.o
In file included from /home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_read_text/ros/src/cob_read_text.cpp:14:0:
/usr/include/pcl-1.7/pcl/ros/conversions.h:44:2: warning: #warning The <pcl/ros/conversions.h> header is deprecated. please use <pcl/conversions.h> instead. [-Wcpp]
 #warning The <pcl/ros/conversions.h> header is deprecated. please use \
  ^
In file included from /usr/include/c++/4.8/backward/strstream:51:0,
                 from /usr/include/vtk-5.8/vtkIOStream.h:112,
                 from /usr/include/vtk-5.8/vtkSystemIncludes.h:40,
                 from /usr/include/vtk-5.8/vtkIndent.h:24,
                 from /usr/include/vtk-5.8/vtkObjectBase.h:43,
                 from /usr/include/vtk-5.8/vtkSmartPointerBase.h:26,
                 from /usr/include/vtk-5.8/vtkSmartPointer.h:23,
                 from /usr/include/pcl-1.7/pcl/visualization/point_cloud_geometry_handlers.h:48,
                 from /usr/include/pcl-1.7/pcl/visualization/point_cloud_handlers.h:41,
                 from /usr/include/pcl-1.7/pcl/visualization/common/actor_map.h:40,
                 from /usr/include/pcl-1.7/pcl/visualization/pcl_visualizer.h:47,
                 from /home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_read_text/ros/src/cob_read_text.cpp:18:
/usr/include/c++/4.8/backward/backward_warning.h:32:2: warning: #warning This file includes at least one deprecated or antiquated header which may be removed without further notice at a future date. Please use a non-deprecated interface with equivalent functionality instead. For a listing of replacement headers and interfaces, consult the file backward_warning.h. To disable this warning use -Wno-deprecated. [-Wcpp]
 #warning \
  ^
Linking CXX executable /home/fxm/projects/indigo/care-o-bot/devel/lib/cob_read_text/cob_read_text
[ 84%] Built target cob_read_text
```

and this one...

```
Scanning dependencies of target object_recording
[ 84%] Building CXX object cob_object_perception/cob_object_recording/CMakeFiles/object_recording.dir/ros/src/object_recording.cpp.o
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_recording/ros/src/object_recording.cpp: In member function ‘tf::Transform ObjectRecording::computeMarkerPose(const ConstPtr&)’:
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_recording/ros/src/object_recording.cpp:418:42: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
  tf::Quaternion mean_orientation(0.,0.,0.);
                                          ^
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_recording/ros/src/object_recording.cpp: In member function ‘void ObjectRecording::computePerspective(const double&, const double&, const double&, tf::Transform&)’:
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_recording/ros/src/object_recording.cpp:522:70: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
  pose1.setRotation(tf::Quaternion(-90./180.*CV_PI, 0., 90./180.*CV_PI));  // rotation in camera direction
                                                                      ^
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_recording/ros/src/object_recording.cpp:524:47: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
  pose2.setRotation(tf::Quaternion(tilt, 0., 0.));  // orientation in tilt direction
                                               ^
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_recording/ros/src/object_recording.cpp:526:46: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
  pose3.setRotation(tf::Quaternion(0., 0., pan));   // orientation in pan direction
                                              ^
Linking CXX executable /home/fxm/projects/indigo/care-o-bot/devel/lib/cob_object_recording/object_recording
[ 84%] Built target object_recording
```

and this one....

```
Scanning dependencies of target object_categorization
[ 82%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/AbstractBlobDetector.cpp.o
[ 82%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/BlobFeature.cpp.o
[ 82%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/BlobList.cpp.o
[ 82%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/DetectorCore.cpp.o
[ 82%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/ICP.cpp.o
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/JBKUtils.cpp.o
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/Math3d.cpp.o
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/ObjectClassifier.cpp.o
In file included from /usr/include/c++/4.8/backward/strstream:51:0,
                 from /usr/include/vtk-5.8/vtkIOStream.h:112,
                 from /usr/include/vtk-5.8/vtkSystemIncludes.h:40,
                 from /usr/include/vtk-5.8/vtkIndent.h:24,
                 from /usr/include/vtk-5.8/vtkObjectBase.h:43,
                 from /usr/include/vtk-5.8/vtkSmartPointerBase.h:26,
                 from /usr/include/vtk-5.8/vtkSmartPointer.h:23,
                 from /usr/include/pcl-1.7/pcl/visualization/point_cloud_geometry_handlers.h:48,
                 from /usr/include/pcl-1.7/pcl/visualization/point_cloud_handlers.h:41,
                 from /usr/include/pcl-1.7/pcl/visualization/common/actor_map.h:40,
                 from /usr/include/pcl-1.7/pcl/visualization/pcl_visualizer.h:47,
                 from /home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_categorization/common/src/ObjectClassifier.cpp:35:
/usr/include/c++/4.8/backward/backward_warning.h:32:2: warning: #warning This file includes at least one deprecated or antiquated header which may be removed without further notice at a future date. Please use a non-deprecated interface with equivalent functionality instead. For a listing of replacement headers and interfaces, consult the file backward_warning.h. To disable this warning use -Wno-deprecated. [-Wcpp]
 #warning \
  ^
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/ros/src/object_categorization.cpp.o
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_categorization/ros/src/object_categorization.cpp: In member function ‘void ObjectCategorization::inputCallback(const ConstPtr&, const ConstPtr&)’:
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_categorization/ros/src/object_categorization.cpp:191:75: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
       pose1.setRotation(tf::Quaternion(0., 90./180.*CV_PI, -90./180.*CV_PI));  // rotation from camera system to object coordinate system
                                                                           ^
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_categorization/ros/src/object_categorization.cpp:193:52: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
       pose2.setRotation(tf::Quaternion(tilt, 0., 0.));  // orientation in tilt direction
                                                    ^
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_categorization/ros/src/object_categorization.cpp:195:52: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
       pose3.setRotation(tf::Quaternion(0., 0., -pan));   // orientation in pan direction
                                                    ^
/home/fxm/projects/indigo/care-o-bot/src/cob_object_perception/cob_object_categorization/ros/src/object_categorization.cpp:198:48: warning: ‘tf::Quaternion::Quaternion(const tfScalar&, const tfScalar&, const tfScalar&)’ is deprecated (declared at /opt/ros/indigo/include/tf/LinearMath/Quaternion.h:51) [-Wdeprecated-declarations]
       pose4.setRotation(tf::Quaternion(0.,0.,0.));
                                                ^
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/OpenCVUtils.cpp.o
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/SharedImageJBK.cpp.o
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/SharedImageSequence.cpp.o
[ 83%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/ThreeDUtils.cpp.o
[ 84%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/timer.cpp.o
[ 84%] Building CXX object cob_object_perception/cob_object_categorization/CMakeFiles/object_categorization.dir/common/src/pcl/rsd.cpp.o
Linking CXX executable /home/fxm/projects/indigo/care-o-bot/devel/lib/cob_object_categorization/object_categorization
[ 84%] Built target object_categorization
```

**Comment by fmessmer** [comment]
Other than that:

I can do a `catkin_make install` without failure!
The dependencies seem to be ok, too...however, my system already is "prepared" for building those packages...
But I cannot give feedback on the capabilities of the individual nodes..this needs to be reviewed by e.g. @ipa-rmb 

@ipa-jsf 
You could set up a jenkins job to test whether all dependencies are specified!
Also, did you test whether all the nodes you want to execute (and launch files and yaml files and so on) are correctly installed? You can check that be sourcing the `install/setup.bash` instead of the `devel/setup.bash`!

File: libzxing/Makefile.tarball (hunk 11 above)
Comment (floweisshardt) [inline libzxing/Makefile.tarball:5]: code.ros.org is still available. I have an account for it. If you like to upload a new tarball version, please let me know

**Comment by fmessmer** [comment]
This PR is insanely long!
What should we do with it? @ipa-rmb can this be merged? and released?

**Comment by ipa-rmb** [comment]
It can be merged. A release is probably possible but some of the programs won't do useful things. Anyways, because of useful and rather stable packages like cob_fiducials it would be worth to release it anyways.

**Comment by floweisshardt** [comment]
merging without testing and indeep review.

**Comment by floweisshardt** [comment]
@ipa-rmb: just found out that you're already in the push group. this means you and @ipa-josh can merge yourself.

**Pull request merged**